aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -1545,12 +1446,19 @@ void qavg_pool3d_nhwc_kernel(
     int padD,
     bool count_include_pad,
     c10::optional<int64_t> divisor_override) {
-  AT_DISPATCH_QINT_TYPES(qx.scalar_type(), "avg_pool3d_nhwc", [&]() {
+  AT_DISPATCH_QINT_TYPES(qx.scalar_type(), fn_name, [&]() {
```

Comment:
nit: would it be possible to move the `AT_DISPATCH_QINT_TYPES` to the parent functions, so that we can get rid of the `fn_name` argument?  Would be more readable and it's worth duplicating one line of code to improve the readability.

Reply:
Yup that's completely fine. I did it like this because it was how `_qadaptive_avg_pool_kernel` did it [here](https://github.com/pytorch/pytorch/blob/fa153184c8f70259337777a1fd1d803c7325f758/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp#L1253).

Reply:
@vkuzo on second thought, I haven't found a way to trivially refactor it so that `AT_DISPATCH_QINT_TYPES` would be in the parent function. The reason is because we pass in `qx.scalar_type()` into the macro `AT_DISPATCH_QINT_TYPES` ([definition here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h#L332)) which then expands another macro `AT_QINT_PRIVATE_CASE_TYPE` ([definition here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/Dispatch.h#L25)). 

Here's the crucial part: in the last macro, it exposes `scalar_type` to the lambda function we pass in. If I moved `AT_DISPATCH_QINT_TYPES` to the parent function, `_qavg_pool_nhwc_kernel` wouldn't know about `scalar_type`. 

One possible way I can see to approach this would be to make the parent's parent (in `q_avgpool.cpp`([here](https://github.com/pytorch/pytorch/blob/2da8c8df0879ec212f1c180b1a11627629742e2c/aten/src/ATen/native/quantized/cpu/q_avgpool.cpp#L199))/`q_avgpool3d.cpp`([here](https://github.com/pytorch/pytorch/blob/2da8c8df0879ec212f1c180b1a11627629742e2c/aten/src/ATen/native/quantized/cpu/q_avgpool3d.cpp#L131)) directly call `_qavg_pool_nhwc_kernel` instead of calling their `2/3d` counterparts that then call `_qavg_pool_nhwc_kernel`.

What do you think?

Reply:
sounds good, doesn't seem worth the trouble then, current version is great!